lms/djangoapps/grades/tasks.py

```diff
@@ -40,17 +42,33 @@
     ValidationError,
     DatabaseNotReadyError,
 )
+POLICY_CHANGE_GRADES_EXPIRY = 60 * 60 * 24
 RECALCULATE_GRADE_DELAY = 2  # in seconds, to prevent excessive _has_db_updated failures. See TNL-6424.
 
 
+def locked_task(expiry_seconds):
+    def task_decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            cache_key = '{}-{}'.format(func.__name__, kwargs['course_key'])
```

Comment:
It would be cleaner to not hard-code `kwargs['course_key']` like this.  I guess we can add a parameter to `locked_task()` called `key` maybe, and pass in 'course_key' when we decorate the task.